### PR TITLE
Sequelize ORM

### PR DESCRIPTION
**Pull request by CalvinRodo**
## Description 
Add Integration for the Sequelize ORM into Sails 

## To Test 
1. run `npm run db:install` to pull down `postgres:latest` image
1. run `npm run db:start` to start the container in detached mode
1. run `npm run db:migrate` to run the migrations in `migrations\`
1. run `sails lift` to start
1. fill our the personal information
1. look in the database for your data

## Note: 
for development purposes you can set `config/models.js` `migrate` to `alter` or `drop` and sails will update the database for you so you don't have to worry about migrations until you have a more stable model

### CODE REVIEW

File: .sailsrc [hunk 1]
```diff
@@ -7,6 +7,9 @@
     "sails-generate": "1.16.13"
   },
   "hooks": {
-    "grunt": false
+    "grunt": false,
+    "orm": false,
```
Comment (jeffmaher) [inline .sailsrc]: Question: What does this disable? Is this what turns off Waterline?

Comment (CalvinRodo) [reply .sailsrc]: yes that's my understanding, this was me basically copying from the instructions in the repo for the Sequelize Hook.

**Comment by jeffmaher** [comment]
Got a 500 error when I clicked `Continue` on the personal page (filled out every field).

```
error: Sending 500 ("Server Error") response: 
 ReferenceError: MedicalReport is not defined
    at Object.store [as personal/store] (/Users/jeff/code/cppd-medical-report-sails/api/controllers/PersonalController.js:60:7)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/router/bind.js:248:46
    at routeTargetFnWrapper (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/router/bind.js:395:9)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:275:10)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:127:14)
    at module.exports (/Users/jeff/code/cppd-medical-report-sails/api/policies/localize.js:6:14)
    at routeTargetFnWrapper (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/router/bind.js:395:9)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:275:10)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:127:14)
    at /Users/jeff/code/cppd-medical-report-sails/api/hooks/validate/index.js:44:18
    at routeTargetFnWrapper (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/router/bind.js:395:9)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:281:22
    at param (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:354:14)
    at param (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:410:3)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:275:10)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:127:14)
    at /Users/jeff/code/cppd-medical-report-sails/api/hooks/routes/index.js:92:28
    at routeTargetFnWrapper (/Users/jeff/code/cppd-medical-report-sails/node_modules/sails/lib/router/bind.js:395:9)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/layer.js:95:5)
    at /Users/jeff/code/cppd-medical-report-sails/node_modules/express/lib/router/index.js:281:22
```

**Comment by CalvinRodo** [comment]
> Got a 500 error when I clicked `Continue` on the personal page (filled out every field).
> 

Did you have the container running and did you run the migration command?

**Comment by jeffmaher** [comment]
I'll try again a little later (I'll also try to pull up the DB itself in a viewer), but here's my quick re-enactment:
https://drive.google.com/a/cds-snc.ca/file/d/19yGHxOQzrEaaCdT_2p0PF_K_fiYprVKf/view?usp=sharing

**Comment by dsamojlenko** [comment]
@jeffmaher @CalvinRodo I know what's going on here ... the model file name needs to be in PascalCase, ie: MedicalReport.js

... Calvin, you probably have already renamed the file locally, but git didn't pick up the change in case. I'll push a commit in a moment with the change that *should* fix this.

**Comment by dsamojlenko** [comment]
@jeffmaher pull latest and give it another shot - double check that your local `/api/models/MedicalReport.js` file has been renamed properly.

**Comment by jeffmaher** [comment]
Thanks @dsamojlenko ! That fixed it. Runs for me now. Confirming that things saved OK.

File: api/controllers/PersonalController.js [hunk 2]
```diff
@@ -58,6 +57,19 @@ module.exports = {
...
+      MedicalReport.create({
+        sin: req.body.social,
+        title: req.body.preferred_title,
+        first_name: req.body.first_name,
+        middle_name: req.body.middle_name,
+        last_name: req.body.last_name,
+        last_name_at_birth: req.body.birth_last_name,
+        birth_date: req.body.birthdate,
+        address: req.body.address,
+        telephone_number: req.body.telephone_number,
```
Comment (dsamojlenko) [inline api/controllers/PersonalController.js]: Should be `telephone` to match 

image

Comment (dsamojlenko) [reply api/controllers/PersonalController.js]: Let's have a chat on making sure column names match input field names and related conventions.

Comment (CalvinRodo) [reply api/controllers/PersonalController.js]: fixed the bug, will rationalize names later

**Pull request merged**